--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -5,8 +5,7 @@
 
 AM_CFLAGS = -w
 AM_LDFLAGS = -lm
-AM_VALAFLAGS = --enable-experimental-non-null   \
-               --metadatadir $(top_srcdir)/deps \
+AM_VALAFLAGS = --metadatadir $(top_srcdir)/deps \
                --vapidir $(top_srcdir)/deps
 
 indicator_keyboard_tests_SOURCES = main.vala   \
--- a/tests/main.vala
+++ b/tests/main.vala
@@ -146,7 +146,7 @@
 	action_group.disconnect (signal_name);
 
 	var state = action_group.get_action_state ("current");
-	var current = ((!) state).get_uint32 ();
+	var current = state.get_uint32 ();
 	assert (current == 2);
 
 	try {
@@ -410,7 +410,7 @@
 	action_group.disconnect (signal_name);
 
 	var state = action_group.get_action_state ("indicator");
-	assert (((!) state).lookup ("visible", "b", out visible));
+	assert (state.lookup ("visible", "b", out visible));
 	assert (visible);
 
 	loop = new MainLoop (null, false);
@@ -433,7 +433,7 @@
 	action_group.disconnect (signal_name);
 
 	state = action_group.get_action_state ("indicator");
-	assert (((!) state).lookup ("visible", "b", out visible));
+	assert (state.lookup ("visible", "b", out visible));
 	assert (!visible);
 
 	loop = new MainLoop (null, false);
@@ -456,7 +456,7 @@
 	action_group.disconnect (signal_name);
 
 	state = action_group.get_action_state ("indicator");
-	assert (((!) state).lookup ("visible", "b", out visible));
+	assert (state.lookup ("visible", "b", out visible));
 	assert (visible);
 }
 
@@ -505,7 +505,7 @@
 	action_group.disconnect (signal_name);
 
 	var state = action_group.get_action_state ("current");
-	var current = ((!) state).get_uint32 ();
+	var current = state.get_uint32 ();
 	assert (current == 1);
 
 	try {
@@ -538,7 +538,7 @@
 	action_group.disconnect (signal_name);
 
 	state = action_group.get_action_state ("current");
-	current = ((!) state).get_uint32 ();
+	current = state.get_uint32 ();
 	assert (current == 0);
 
 	try {
